### PR TITLE
DOC: point source links to github

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
This PR introduces `sphinx.ext.linkcode` to our docs. This allows us to point the `[source]` link in the API docs directly to the relevant code on GitHub (with highlighting of the respective function 🚀 ).

### CODE REVIEW

**Comment by FirefoxMetzger** [comment]
@almarklein RTR :) lots of reviews this week.

File: docs/conf.py [hunk 1]
```diff
@@ -118,6 +123,54 @@
 # modindex_common_prefix = []
 
 
+# Function to generate source links pointing to GitHub
+def linkcode_resolve(domain, info):
```
Comment (almarklein) [inline docs/conf.py:127]: Is this function magically picked up by linkcode?

**Review by almarklein — APPROVED** [review]

File: docs/conf.py (hunk 1 above)
Comment (FirefoxMetzger) [reply docs/conf.py:127]: `sphinx.ext.linkcode` looks for a function called `linkcode_resolve` in the sphinx config (`conf-py`) and uses it as a callback to figure out where the source hyperlink should point to, yes.

You can try it out in the PR docs, e.g. for [imread](https://imageio--834.org.readthedocs.build/en/834/_autosummary/imageio.v3.imread.html#imageio.v3.imread) and click on the `[Source]` button next to the name of the function.

Comment (almarklein) [reply docs/conf.py:127]: Nice! 🚀

**Pull request merged**